### ISSUE TITLE
Inconsistent behavior for vertical_parameter_alignment_on_call rule

### ISSUE THREAD

**Issue by steverichey** [realm/SwiftLint#2151]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

False positive for `vertical_parameter_alignment_on_call` rule when function takes pair variadic arguments.

```bash
.../example.swift:10:9: warning: Vertical Parameter Alignment On Call Violation: Function parameters should be aligned vertically if they're in multiple lines in a method call. (vertical_parameter_alignment_on_call)
.../example.swift:11:9: warning: Vertical Parameter Alignment On Call Violation: Function parameters should be aligned vertically if they're in multiple lines in a method call. (vertical_parameter_alignment_on_call)
.../example.swift:12:9: warning: Vertical Parameter Alignment On Call Violation: Function parameters should be aligned vertically if they're in multiple lines in a method call. (vertical_parameter_alignment_on_call)
Done linting! Found 3 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version: *0.25.1*
* Installation method: *Homebrew*

```yml
opt_in_rules:
  - vertical_parameter_alignment_on_call
```

* No nested configurations
* Xcode 9.3

```swift
func doThingA() {
    let one = 1
    let two = 2
    let three = 3
    let four = 4

    // triggers
    doThingB(with:
        one,
        two
    )

    // does not trigger
    doThingC(
        one,
        two
    )
}

func doThingB(with: (Int, Int)...) {}

func doThingC(_ with: (Int, Int)...) {}
```

**Comment by marcelofabri** [comment]
This is not related to variadic parameters, but whether the first parameter is named or not.

```swift
// triggers
doThingB(with:
    one,
    two
)

// does not trigger
doThingC(
    one,
    two
)
```

This also triggers the issue.

However, I'm not sure this is actually a false positive. Maybe there could be a configuration to allow this pattern. 

@steverichey Would you like to work on that? 😊

**Comment by steverichey** [comment]
I've updated the description with the simpler example and clarified in the title that this is "inconsistent" rather than a false positive. I'd be happy to work on this when I have time, but I don't know when that might be. Feel free to close if this is expected behavior.

**Comment by rlasantenuance** [comment]
So is the expected behavior that if a function has a label the label and all values should be on the same line? It works, but not sure I'd agree that's cleaner than either of the two versions that allow for distinguishing the parameter label from the items passed.

```
// triggers
doThingB(with:
    one,
    two
)

// triggers
doThingB(
    with:
        one,
        two
)

// does not triggers
doThingB(
    with:
    one,
    two
)
```
